app/models/user.rb

```diff
@@ -0,0 +1,59 @@
+# Represents a single application user.
+class User < ApplicationRecord
+  # Accessors
+  attr_reader :email, :password
+
+  def email=(value)
+    super(value)
+    write_attribute(:email_hash, value) # Need hash method
+  end
+
+  def password=(value)
+    super(value)
+    write_attribute(:password_hash, value) # Need hash method
+  end
+
+  # Associations
+  has_and_belongs_to_many :groups
+  has_many :lists
+
+  # Validations
+  validates :email_hash, presence: true,
+                         uniqueness: { case_sensitive: false }
+  validates :password_hash, presence: true
+
+  # Callbacks
+  before_destroy :delete_owned_lists
+
+  # Performs additional initialization when creating a new instance
+  # of the User class. Required to allow for the instance to be
+  # created using content not stored in the database.
+  def initialize(email, password)
+    super()
+    @email = email
+    write_attribute(:email_hash, email) # Need hash method
+    @password = password
+    write_attribute(:password_hash, password) # Need hash method
+  end
+
+  # Override of User.create to allow for the instance to be created
+  # using content not stored in the database.
+  def self.create(email, password)
```

Comment:
What's this override for?

Reply:
The initialize method is only called on User.new. Calling user.create requires the email and password hashes to be provided in order to create the record. The override allows for the functionality of the create method without requiring the hashes to already be generated (looking to handle that within the model, as we only really care about it for data storage).